### ISSUE TITLE
Don't use the click event to detect when a checkbox is checked or unchecked

### ISSUE THREAD

**Issue by emilyanew** [vanilla/vanilla#2777]
The use of click() vs change() is inconsistent throughout the core vanilla javascript.

The correct way to detect a checkbox check or uncheck event is to use change() and not click().

The following files detect checkbox check events using click():

discussions.js
discussion.js
dashboard/js/log.js
dashboard/js/profile.js

The following files detect checkbox check events using the correct change() event:

js/global.js

Why does this matter? Well, many themes (mine included) use icheck to style checkboxes and other elements. Hijacking the click event within vanilla causes these libraries to fail.

Please use the correct change event for checkboxes.

**Comment by linc** [comment]
Gonna raise this for @slafleche and @Charrondev to comment on or take into consideration for the revamp.

**Comment by slafleche** [comment]
@Charrondev put that on your checklist for the JS refactoring.

As for future markup, we'll be using the "label" trick in the future to allow styling on checkboxes without the need to use a plugin and without needing JS.

**Comment by charrondev** [comment]
This is fixed in our new components. We have no current changes to alter these older components to integrate these improvements due to backwards compatibility concerns w/ existing themes.

Eventually we will rebuild the pages w/ new components & some feature flag to cover the BC break.

Until then see the `<CheckBox />`,  `<RadioButton />`, `<DashboardCheckBox />`, & `DashboardRadioButton />` components.
